vfs/read.go

```diff
@@ -225,10 +237,22 @@ func (fh *ReadFileHandle) readAt(p []byte, off int64) (n int, err error) {
 			// }
 			if err == nil {
 				break
-			} else if (err == io.ErrUnexpectedEOF || err == io.EOF) && newOffset == fh.size {
-				// Have read to end of file - reset error
-				err = nil
-				break
+			} else if err == io.ErrUnexpectedEOF || err == io.EOF {
+				if newOffset == fh.size {
+					// Have read to end of file - reset error
+					err = nil
+					break
+				} else if fh.chunkSize > 0 && newOffset == fh.chunkOffset+fh.chunkSize {
+					// Have read to end of chunk - reset error and reread
+					err = nil
+					doSeek = true
+					doReopen = true
+					off = newOffset
+					p = p[n:]
+					fh.chunkSize *= 2
```

Comment:
Why do you think doubling the chunk size each time is a good idea?  Maybe it should have a maximum?

Reply:
The intention is to keep the number of requests small. A file being read for many chunks continuously is probably being read to further.
Most providers don't provide information, if the amount of "downloaded bytes" or the "requested bytes" are limited.
For Google Drive i read a value of 10TB downloads per day. From my experience they seem to count the requested bytes, because i could download parts of large files, after the whole file download was returning "403 Access Denied".
Probably most providers count "requested bytes", but there could be other limits as well.

Maybe a second option `--vfs-read-chunk-growth` could give this decision to the user, where the default is to keep the chunk size and with the option enabled chunk size doubles.

Reply:
Requested bytes are the easy thing to measure for the cloud provider.  Downloaded bytes are tricker to measure...

An option for the user to control the doubling sounds like a good idea to trade latency for efficiency if required.

Do you think the buffer size should be reset to the default on seeks too?

Reply:
Resetting the size on seek sounds good.I will add that.